lightning/src/ln/onion_utils.rs

```diff
@@ -418,12 +418,20 @@ pub(super) fn construct_onion_packet(
 #[allow(unused)]
 pub(super) fn construct_trampoline_onion_packet(
 	payloads: Vec<msgs::OutboundTrampolinePayload>, onion_keys: Vec<OnionKeys>,
-	prng_seed: [u8; 32], associated_data: &PaymentHash, length: u16,
+	prng_seed: [u8; 32], associated_data: &PaymentHash,
 ) -> Result<msgs::TrampolineOnionPacket, ()> {
-	let mut packet_data = vec![0u8; length as usize];
+	let packet_length: usize = payloads
+		.iter()
+		.map(|p| {
+			let mut payload_len = LengthCalculatingWriter(0);
+			p.write(&mut payload_len).expect("Failed to calculate length");
+			payload_len.0 + 32
+		})
+		.sum();
```

Comment:
Is it harmful for privacy to not use a fixed length?

Reply:
Would the code be more confusing if we padded after the fact? I'd kinda prefer to review this commit amidst some more context.